mlguess/keras/models.py

```diff
@@ -1241,27 +1268,24 @@ def predict_ensemble(self, x, weight_locations, batch_size=None):
             predictions = np.empty((num_models,) + (x.shape[0],) + output_shape)
 
         # Predict for the remaining models
-        for i, weight_location in enumerate(weight_locations[1:]):
+        for i, weight_location in enumerate(self.ensemble_member_files[1:]):
             model_instance = self.model
             model_instance.load_weights(weight_location)
             y_prob = model_instance.predict(x, batch_size=batch_size)
             predictions[i + 1] = y_prob
 
         return predictions
 
-    def compute_uncertainties(self, y_pred, num_classes=4):
-        return calc_prob_uncertainty(y_pred, num_classes=num_classes)
-
-
-def calc_prob_uncertainty(y_pred, num_classes=4):
-    evidence = tf.nn.relu(y_pred)
-    alpha = evidence + 1
-    S = tf.keras.backend.sum(alpha, axis=1, keepdims=True)
-    u = num_classes / S
-    prob = alpha / S
-    epistemic = prob * (1 - prob) / (S + 1)
-    aleatoric = prob - prob**2 - epistemic
-    return prob, u, aleatoric, epistemic
+    def predict_uncertainty(self, x, num_classes=4):
```

Comment:
You should be able to get num_classes from y_pred.shape[1] rather than having the user specify it and potentially get an incorrect result if they forget to change the default.